utils/brc20.ts

```diff
@@ -0,0 +1,74 @@
+export const isValidContentType = (contentType: string) =>
+  contentType.startsWith('text/plain') || contentType.startsWith('application/json');
+
+export const isValidFields = (parsedFields: string[], requiredFields: Set<string>, optionalFields?: Set<string>) =>
+  parsedFields.every((f) => requiredFields.has(f) || optionalFields?.has(f)) &&
+  [...requiredFields].every((f) => parsedFields.includes(f));
+
+const isNumber = (value: string) => !Number.isNaN(Number(value));
+
+export type Brc20Definition = {
+  op: 'deploy' | 'mint' | 'transfer';
+  tick: string;
+  value: string;
+};
+
+export const getBrc20Details = (content: string, contentType: string): undefined | Brc20Definition => {
```

Comment:
quick question. do we want to write a few UT on this?

Reply:
absolutely

Reply:
done @fedeerbes. thanks for encouraging 😄